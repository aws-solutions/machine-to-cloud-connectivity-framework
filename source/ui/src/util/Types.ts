// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
// SPDX-License-Identifier: Apache-2.0

import React from 'react';

export type ErrorMessage = { errorMessage: string };
export type FormControlElement = HTMLInputElement | HTMLSelectElement | HTMLTextAreaElement;
export type KeyStringValue = Record<string, string | undefined>;
type ApiMethod = 'get' | 'post';
type GreengrassCoreDeviceStatus = 'HEALTHY' | 'UNHEALTHY';
type LoggingLevel = 'ERROR' | 'WARN' | 'INFO' | 'DEBUG' | 'VERBOSE';

export enum ConnectionControl {
  DEPLOY = 'deploy',
  START = 'start',
  STOP = 'stop',
  UPDATE = 'update',
  DELETE = 'delete',
  PUSH = 'push',
  PULL = 'pull'
}

export enum LogType {
  INFO = 'info',
  ERROR = 'error'
}

export enum MachineProtocol {
  OPCDA = 'opcda',
  OPCUA = 'opcua',
  OSIPI = 'osipi'
}

export enum MessageModalType {
  MESSAGE,
  CONFIRM
}

export enum PaginationType {
  PREV,
  NEXT
}

export enum CreatedBy {
  SYSTEM = 'System',
  USER = 'User'
}

export enum GreengrassCoreDeviceControl {
  CREATE = 'create',
  DELETE = 'delete'
}

export enum OsiPiAuthMode {
  ANONYMOUS = 'ANONYMOUS',
  BASIC = 'BASIC'
  //TODO: Figure out how to support Kerberos auth
  // KERBEROS = 'KERBEROS'
}

export interface AmplifyConfigurationInput {
  apiEndpoint: string;
  identityPoolId: string;
  loggingLevel: LoggingLevel;
  region: string;
  s3Bucket: string;
  userPoolId: string;
  webClientId: string;
}

export interface AmplifyConfigurationResponse {
  API: {
    endpoints: { name: string; endpoint: string; region: string }[];
  };
  Auth: {
    identityPoolId: string;
    region: string;
    userPoolId: string;
    userPoolWebClientId: string;
  };
  Storage: {
    AWSS3: {
      bucket: string;
      region: string;
    };
  };
}

export interface ConnectionDefinition {
  control: ConnectionControl;
  connectionName: string;
  protocol: MachineProtocol;
  area?: string;
  greengrassCoreDeviceName?: string;
  machineName?: string;
  opcDa?: OpcDaDefinition;
  opcUa?: OpcUaDefinition;
  osiPi?: OsiPiDefinition;
  process?: string;
  sendDataToIoTSiteWise?: boolean;
  sendDataToIoTTopic?: boolean;
  sendDataToKinesisDataStreams?: boolean;
  sendDataToTimestream?: boolean;
  siteName?: string;
  logLevel?: string;
}

export interface CommonDefinition {
  machineIp: string;
  serverName: string;
}

export interface OpcDaDefinition extends CommonDefinition {
  interval: number | string;
  iterations: number | string;
  listTags?: string[];
  tags?: string[];
}

export interface OpcUaDefinition extends CommonDefinition {
  port?: number | string;
}

export interface OsiPiDefinition {
  apiUrl: string;
  serverName: string;
  authMode: OsiPiAuthMode;
  verifySSL: boolean;
  username?: string;
  password?: string;
  tags: string[];
  requestFrequency: number | string;
  catchupFrequency: number | string;
  maxRequestDuration: number | string;
  queryOffset: number | string;
}

export interface ListConnectionsItem {
  connectionName: string;
  machineName: string;
  protocol: MachineProtocol;
  status: ConnectionControl;
  sendDataToIoTSiteWise: boolean;
  sendDataToIoTTopic: boolean;
  sendDataToKinesisDataStreams: boolean;
}

export interface ListConnectionsResponse {
  connections: ListConnectionsItem[];
  nextToken?: string;
}

export interface ControlConnectionResponse {
  connectionName: string;
  control: ConnectionControl;
  result: string;
}

export interface CreateUpdateConnectionResponse {
  connectionName: string;
  control: ConnectionControl.DEPLOY | ConnectionControl.UPDATE;
  message: string;
}

export interface ModalProps {
  show: boolean;
  hide: () => void;
}

export interface ConnectionLogsModalProps extends ModalProps {
  connectionName: string;
}

export interface MessageModalProps extends ModalProps {
  message: string | React.ReactNode;
  modalType: MessageModalType;
  confirmAction?: () => void | Promise<void>;
}

export interface GetConnectionResponse {
  connectionName: string;
  greengrassCoreDeviceName: string;
  protocol: MachineProtocol;
  sendDataToIoTSiteWise: boolean;
  sendDataToIoTTopic: boolean;
  sendDataToKinesisDataStreams: boolean;
  sendDataToTimestream: boolean;
  area?: string;
  machineName?: string;
  logLevel?: string;
  opcDa?: OpcDaDefinition;
  opcUa?: OpcUaDefinition;
  osiPi?: OsiPiDefinition;
  process?: string;
  siteName?: string;
  opcDaListTags?: string;
  opcDaTags?: string;
  osiPiTags?: string;
}

export interface FormProps {
  connection: GetConnectionResponse;
  onChange: (event: React.ChangeEvent<FormControlElement>) => void;
  errors: KeyStringValue;
}

export interface OpcUaFormProps extends FormProps {
  region: string;
}

export interface ListLogsItem {
  connectionName: string;
  timestamp: number;
  logType: LogType;
  message: string;
}

export interface ListLogsResponse {
  logs: ListLogsItem[];
  nextToken?: string;
}

export interface AlphaNumericValidationProps {
  value: string;
  maxLength: number;
  errors: KeyStringValue;
  errorKeyName: string;
  errorMessage: string;
}

export interface ErrorProps {
  message?: string;
  response?: {
    data: ErrorMessage | unknown;
  };
}

export interface ApiProps {
  method: ApiMethod;
  path: string;
  options?: Record<string, unknown>;
}

export interface ListGreengrassCoreDevicesResponse {
  greengrassCoreDevices: ListGreengrassCoreDevicesItem[];
  nextToken?: string;
}

export interface ListGreengrassCoreDevicesItem {
  name: string;
  createdBy: CreatedBy;
  numberOfConnections: number;
}

export interface UserGreengrassCoreDeviceResponse {
  greengrassCoreDevices: UserGreengrassCoreDevice[];
}

export interface UserGreengrassCoreDevice {
  coreDeviceThingName: string;
  status: GreengrassCoreDeviceStatus;
}

export interface GreengrassCoreDevicePostResponse {
  name: string;
  message: string;
}

export interface GreengrassCoreDeviceDomIds {
  category: string;
  greengrassCoreDeviceName: string;
}
