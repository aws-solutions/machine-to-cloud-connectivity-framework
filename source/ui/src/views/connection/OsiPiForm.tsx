// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
// SPDX-License-Identifier: Apache-2.0

import { I18n } from '@aws-amplify/core';
import Form from 'react-bootstrap/Form';
import Card from 'react-bootstrap/Card';
import { FormProps, OsiPiAuthMode } from '../../util/types';
import EmptyRow from '../../components/EmptyRow';

/**
 * Renders the OSI PI form.
 * @param props The properties for the OSI PI form
 * @returns The OSI PI form
 */
export default function OsiPiForm(props: FormProps): JSX.Element {
  const { connection, onChange, errors } = props;

  return (
    <>
      <Form.Group>
        <Form.Label>
          {I18n.get('api.url')} <span className="red-text">*</span>
        </Form.Label>
        <Form.Text muted>{I18n.get('description.osiPi.apiUrl')}</Form.Text>
        <Form.Control
          id="osiPi_apiUrl"
          required
          type="text"
          placeholder={I18n.get('placeholder.osiPi.apiUrl')}
          defaultValue={connection.osiPi?.apiUrl}
          onChange={onChange}
          isInvalid={errors.osiPi_apiUrl != undefined}
        />
        <Form.Control.Feedback type="invalid">{errors.osiPi_apiUrl}</Form.Control.Feedback>
      </Form.Group>
      <Form.Group>
        <Form.Check
          inline
          type="checkbox"
          id="osiPi_verifySSL"
          label={I18n.get('verify.ssl')}
          checked={connection.osiPi?.verifySSL}
          onChange={onChange}
        />
      </Form.Group>
      <Form.Group>
        <Form.Label>
          {I18n.get('server.name')} <span className="red-text">*</span>
        </Form.Label>
        <Form.Text muted>{I18n.get('description.osiPi.serverName')}</Form.Text>
        <Form.Control
          id="osiPi_serverName"
          required
          type="text"
          placeholder={I18n.get('placeholder.osiPi.serverName')}
          defaultValue={connection.osiPi?.serverName}
          onChange={onChange}
          isInvalid={!!errors.osiPi_serverName}
        />
        <Form.Control.Feedback type="invalid">{errors.osiPi_serverName}</Form.Control.Feedback>
      </Form.Group>
      <Card>
        <Card.Body>
          <Card.Subtitle>{I18n.get('authentication')}</Card.Subtitle>
          <EmptyRow />
          <Form.Group>
            <Form.Label>
              {I18n.get('auth.mode')} <span className="red-text">*</span>
            </Form.Label>
            <Form.Text muted>{I18n.get('description.osiPi.authMode')}</Form.Text>
            <Form.Control id="osiPi_authMode" as="select" onChange={onChange} value={connection.osiPi?.authMode}>
              <option value={OsiPiAuthMode.ANONYMOUS}>Anonymous</option>
              <option value={OsiPiAuthMode.BASIC}>Basic</option>
            </Form.Control>
          </Form.Group>

          {connection.osiPi?.authMode === OsiPiAuthMode.BASIC && (
            <Form.Group>
              <Form.Label>
                {I18n.get('username')} <span className="red-text">*</span>
              </Form.Label>
              <Form.Text muted>{I18n.get('description.osiPi.username')}</Form.Text>
              <Form.Control
                id="osiPi_username"
                required
                type="text"
                placeholder={I18n.get('placeholder.osiPi.username')}
                defaultValue={connection.osiPi?.username}
                onChange={onChange}
                isInvalid={!!errors.osiPi_username}
              />
              <Form.Control.Feedback type="invalid">{errors.osiPi_username}</Form.Control.Feedback>
            </Form.Group>
          )}

          {connection.osiPi?.authMode === OsiPiAuthMode.BASIC && (
            <Form.Group>
              <Form.Label>
                {I18n.get('password')} <span className="red-text">*</span>
              </Form.Label>
              <Form.Text muted>{I18n.get('description.osiPi.password')}</Form.Text>
              <Form.Control
                id="osiPi_password"
                required
                type="password"
                placeholder={I18n.get('placeholder.osiPi.password')}
                defaultValue={connection.osiPi?.password}
                onChange={onChange}
                isInvalid={!!errors.osiPi_password}
              />
              <Form.Control.Feedback type="invalid">{errors.osiPi_password}</Form.Control.Feedback>
            </Form.Group>
          )}
        </Card.Body>
      </Card>

      <EmptyRow />

      <Card>
        <Card.Body>
          <Card.Subtitle>{I18n.get('query.config')}</Card.Subtitle>
          <EmptyRow />
          <Form.Group>
            <Form.Label>
              {I18n.get('request.frequency')} <span className="red-text">*</span>
            </Form.Label>
            <Form.Text muted>{I18n.get('description.osiPi.requestFrequency')}</Form.Text>
            <Form.Control
              id="osiPi_requestFrequency"
              required
              type="text"
              placeholder={I18n.get('placeholder.osiPi.requestFrequency')}
              defaultValue={connection.osiPi?.requestFrequency}
              onChange={onChange}
              isInvalid={!!errors.osiPi_requestFrequency}
            />
            <Form.Control.Feedback type="invalid">{errors.osiPi_requestFrequency}</Form.Control.Feedback>
          </Form.Group>

          <Form.Group>
            <Form.Label>
              {I18n.get('catchup.frequency')} <span className="red-text">*</span>
            </Form.Label>
            <Form.Text muted>{I18n.get('description.osiPi.catchupFrequency')}</Form.Text>
            <Form.Control
              id="osiPi_catchupFrequency"
              required
              type="text"
              placeholder={I18n.get('placeholder.osiPi.catchupFrequency')}
              defaultValue={connection.osiPi?.catchupFrequency}
              onChange={onChange}
              isInvalid={!!errors.osiPi_catchupFrequency}
            />
            <Form.Control.Feedback type="invalid">{errors.osiPi_catchupFrequency}</Form.Control.Feedback>
          </Form.Group>

          <Form.Group>
            <Form.Label>
              {I18n.get('max.request.duration')} <span className="red-text">*</span>
            </Form.Label>
            <Form.Text muted>{I18n.get('description.osiPi.maxRequestDuration')}</Form.Text>
            <Form.Control
              id="osiPi_maxRequestDuration"
              required
              type="text"
              placeholder={I18n.get('placeholder.osiPi.maxRequestDuration')}
              defaultValue={connection.osiPi?.maxRequestDuration}
              onChange={onChange}
              isInvalid={!!errors.osiPi_maxRequestDuration}
            />
            <Form.Control.Feedback type="invalid">{errors.osiPi_maxRequestDuration}</Form.Control.Feedback>
          </Form.Group>

          <Form.Group>
            <Form.Label>
              {I18n.get('query.offset')} <span className="red-text">*</span>
            </Form.Label>
            <Form.Text muted>{I18n.get('description.osiPi.queryOffset')}</Form.Text>
            <Form.Control
              id="osiPi_queryOffset"
              required
              type="text"
              placeholder={I18n.get('placeholder.osiPi.queryOffset')}
              defaultValue={connection.osiPi?.queryOffset}
              onChange={onChange}
              isInvalid={!!errors.osiPi_queryOffset}
            />
            <Form.Control.Feedback type="invalid">{errors.osiPi_queryOffset}</Form.Control.Feedback>
          </Form.Group>
        </Card.Body>
      </Card>

      <EmptyRow />

      <Card>
        <Card.Body>
          <Card.Subtitle>
            {I18n.get('tags')} {I18n.get('tags.explanation')}
          </Card.Subtitle>
          <EmptyRow />
          <Form.Group>
            <Form.Label>{I18n.get('read.tags')}</Form.Label>
            <Form.Control
              id="osiPiTags"
              as="textarea"
              placeholder={I18n.get('placeholder.osiPi.tags')}
              onChange={onChange}
              defaultValue={connection.osiPiTags}
              isInvalid={!!errors.osiPi_Tags}
            />
            <Form.Control.Feedback type="invalid">{errors.osiPi_Tags}</Form.Control.Feedback>
          </Form.Group>
        </Card.Body>
      </Card>
    </>
  );
}
