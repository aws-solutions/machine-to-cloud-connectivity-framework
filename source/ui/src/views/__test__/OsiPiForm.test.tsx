// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
// SPDX-License-Identifier: Apache-2.0

import { I18n } from '@aws-amplify/core';
import { render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import OsiPiForm from '../connection/OsiPiForm';
import { GetConnectionResponse, MachineProtocol, OsiPiAuthMode } from '../../util/types';
import { INIT_CONNECTION } from '../../util/utils';

test('renders the default OsiPiForm', async () => {
  const connection: GetConnectionResponse = INIT_CONNECTION;
  const osiPiForm = render(<OsiPiForm connection={connection} onChange={() => console.log('changed')} errors={{}} />);
  expect(osiPiForm.container).toMatchSnapshot();
});

test('renders the OsiPiForm with the existing connection', async () => {
  const connection: GetConnectionResponse = {
    connectionName: 'mock-connection-id',
    greengrassCoreDeviceName: 'mock-greengrass-device-name',
    protocol: MachineProtocol.OSIPI,
    sendDataToIoTSiteWise: true,
    sendDataToIoTTopic: true,
    sendDataToKinesisDataStreams: true,
    sendDataToTimestream: true,
    sendDataToHistorian: true,
    area: 'mock-area',
    machineName: 'mock-machine',
    osiPi: {
      apiUrl: 'https://osiPiServer/piwebapi',
      verifySSL: true,
      serverName: 'mock-server',
      authMode: OsiPiAuthMode.BASIC,
      username: 'mock-user',
      password: 'mock-password',
      requestFrequency: 5,
      catchupFrequency: 0.1,
      maxRequestDuration: 60,
      queryOffset: 0,
      tags: ['tag1', 'tag2']
    },
    osiPiTags: 'tag1\ntag2'
  };

  const osiPiForm = render(<OsiPiForm connection={connection} onChange={() => console.log('changed')} errors={{}} />);
  expect(osiPiForm.container).toMatchSnapshot();
});

test('renders the OsiPiForm errors', async () => {
  const connection: GetConnectionResponse = INIT_CONNECTION;
  const errors = {
    connectionName: 'connectionName invalid',
    sendDataTo: 'sendDataTo invalid',
    apiUrl: 'apiUrl invalid',
    serverName: 'serverName invalid',
    username: 'username invalid',
    password: 'password invalid',
    requestFrequency: 'requestFrequency invalid',
    catchupFrequency: 'maxCatchupFrequency invalid',
    maxRequestDuration: 'maxRequestDuration invalid',
    queryOffset: 'queryOffset invalid',
    tags: 'tags invalid'
  };
  const osiPiForm = render(
    <OsiPiForm connection={connection} onChange={() => console.log('changed')} errors={errors} />
  );
  expect(osiPiForm.container).toMatchSnapshot();
});

test('tests handleValueChange function', async () => {
  const connection: GetConnectionResponse = INIT_CONNECTION;
  const osiPiForm = render(<OsiPiForm connection={connection} onChange={event => console.log(event)} errors={{}} />);
  const input = osiPiForm.getByPlaceholderText(I18n.get('placeholder.osiPi.requestFrequency'));
  userEvent.type(input, '{backspace}60');
  expect((input as HTMLInputElement).value).toEqual('60');
});
