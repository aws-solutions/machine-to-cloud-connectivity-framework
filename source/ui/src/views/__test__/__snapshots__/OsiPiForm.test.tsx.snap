// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the OsiPiForm errors 1`] = `
<div>
  <div
    class="form-group"
  >
    <label
      class="form-label"
    >
      api.url
       
      <span
        class="red-text"
      >
        *
      </span>
    </label>
    <small
      class="form-text text-muted"
    >
      description.osiPi.apiUrl
    </small>
    <input
      class="form-control"
      id="osiPi_apiUrl"
      placeholder="placeholder.osiPi.apiUrl"
      required=""
      type="text"
      value=""
    />
    <div
      class="invalid-feedback"
    />
  </div>
  <div
    class="form-group"
  >
    <div
      class="form-check form-check-inline"
    >
      <input
        checked=""
        class="form-check-input"
        id="osiPi_verifySSL"
        type="checkbox"
      />
      <label
        class="form-check-label"
        for="osiPi_verifySSL"
        title=""
      >
        verify.ssl
      </label>
    </div>
  </div>
  <div
    class="form-group"
  >
    <label
      class="form-label"
    >
      server.name
       
      <span
        class="red-text"
      >
        *
      </span>
    </label>
    <small
      class="form-text text-muted"
    >
      description.osiPi.serverName
    </small>
    <input
      class="form-control"
      id="osiPi_serverName"
      placeholder="placeholder.osiPi.serverName"
      required=""
      type="text"
      value=""
    />
    <div
      class="invalid-feedback"
    />
  </div>
  <div
    class="card"
  >
    <div
      class="card-body"
    >
      <div
        class="card-subtitle h6"
      >
        authentication
      </div>
      <div
        class="row"
      >
        <div
          class="col"
        >
          <span>
             
          </span>
        </div>
      </div>
      <div
        class="form-group"
      >
        <label
          class="form-label"
        >
          auth.mode
           
          <span
            class="red-text"
          >
            *
          </span>
        </label>
        <small
          class="form-text text-muted"
        >
          description.osiPi.authMode
        </small>
        <select
          class="form-control"
          id="osiPi_authMode"
        >
          <option
            value="ANONYMOUS"
          >
            Anonymous
          </option>
          <option
            value="BASIC"
          >
            Basic
          </option>
        </select>
      </div>
    </div>
  </div>
  <div
    class="row"
  >
    <div
      class="col"
    >
      <span>
         
      </span>
    </div>
  </div>
  <div
    class="card"
  >
    <div
      class="card-body"
    >
      <div
        class="card-subtitle h6"
      >
        query.config
      </div>
      <div
        class="row"
      >
        <div
          class="col"
        >
          <span>
             
          </span>
        </div>
      </div>
      <div
        class="form-group"
      >
        <label
          class="form-label"
        >
          request.frequency
           
          <span
            class="red-text"
          >
            *
          </span>
        </label>
        <small
          class="form-text text-muted"
        >
          description.osiPi.requestFrequency
        </small>
        <input
          class="form-control"
          id="osiPi_requestFrequency"
          placeholder="placeholder.osiPi.requestFrequency"
          required=""
          type="text"
          value="5"
        />
        <div
          class="invalid-feedback"
        />
      </div>
      <div
        class="form-group"
      >
        <label
          class="form-label"
        >
          catchup.frequency
           
          <span
            class="red-text"
          >
            *
          </span>
        </label>
        <small
          class="form-text text-muted"
        >
          description.osiPi.catchupFrequency
        </small>
        <input
          class="form-control"
          id="osiPi_catchupFrequency"
          placeholder="placeholder.osiPi.catchupFrequency"
          required=""
          type="text"
          value="0.1"
        />
        <div
          class="invalid-feedback"
        />
      </div>
      <div
        class="form-group"
      >
        <label
          class="form-label"
        >
          max.request.duration
           
          <span
            class="red-text"
          >
            *
          </span>
        </label>
        <small
          class="form-text text-muted"
        >
          description.osiPi.maxRequestDuration
        </small>
        <input
          class="form-control"
          id="osiPi_maxRequestDuration"
          placeholder="placeholder.osiPi.maxRequestDuration"
          required=""
          type="text"
          value="60"
        />
        <div
          class="invalid-feedback"
        />
      </div>
      <div
        class="form-group"
      >
        <label
          class="form-label"
        >
          query.offset
           
          <span
            class="red-text"
          >
            *
          </span>
        </label>
        <small
          class="form-text text-muted"
        >
          description.osiPi.queryOffset
        </small>
        <input
          class="form-control"
          id="osiPi_queryOffset"
          placeholder="placeholder.osiPi.queryOffset"
          required=""
          type="text"
          value="0"
        />
        <div
          class="invalid-feedback"
        />
      </div>
    </div>
  </div>
  <div
    class="row"
  >
    <div
      class="col"
    >
      <span>
         
      </span>
    </div>
  </div>
  <div
    class="card"
  >
    <div
      class="card-body"
    >
      <div
        class="card-subtitle h6"
      >
        tags
         
        tags.explanation
      </div>
      <div
        class="row"
      >
        <div
          class="col"
        >
          <span>
             
          </span>
        </div>
      </div>
      <div
        class="form-group"
      >
        <label
          class="form-label"
        >
          read.tags
        </label>
        <textarea
          class="form-control"
          id="osiPiTags"
          placeholder="placeholder.osiPi.tags"
        />
        <div
          class="invalid-feedback"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`renders the OsiPiForm with the existing connection 1`] = `
<div>
  <div
    class="form-group"
  >
    <label
      class="form-label"
    >
      api.url
       
      <span
        class="red-text"
      >
        *
      </span>
    </label>
    <small
      class="form-text text-muted"
    >
      description.osiPi.apiUrl
    </small>
    <input
      class="form-control"
      id="osiPi_apiUrl"
      placeholder="placeholder.osiPi.apiUrl"
      required=""
      type="text"
      value="https://osiPiServer/piwebapi"
    />
    <div
      class="invalid-feedback"
    />
  </div>
  <div
    class="form-group"
  >
    <div
      class="form-check form-check-inline"
    >
      <input
        checked=""
        class="form-check-input"
        id="osiPi_verifySSL"
        type="checkbox"
      />
      <label
        class="form-check-label"
        for="osiPi_verifySSL"
        title=""
      >
        verify.ssl
      </label>
    </div>
  </div>
  <div
    class="form-group"
  >
    <label
      class="form-label"
    >
      server.name
       
      <span
        class="red-text"
      >
        *
      </span>
    </label>
    <small
      class="form-text text-muted"
    >
      description.osiPi.serverName
    </small>
    <input
      class="form-control"
      id="osiPi_serverName"
      placeholder="placeholder.osiPi.serverName"
      required=""
      type="text"
      value="mock-server"
    />
    <div
      class="invalid-feedback"
    />
  </div>
  <div
    class="card"
  >
    <div
      class="card-body"
    >
      <div
        class="card-subtitle h6"
      >
        authentication
      </div>
      <div
        class="row"
      >
        <div
          class="col"
        >
          <span>
             
          </span>
        </div>
      </div>
      <div
        class="form-group"
      >
        <label
          class="form-label"
        >
          auth.mode
           
          <span
            class="red-text"
          >
            *
          </span>
        </label>
        <small
          class="form-text text-muted"
        >
          description.osiPi.authMode
        </small>
        <select
          class="form-control"
          id="osiPi_authMode"
        >
          <option
            value="ANONYMOUS"
          >
            Anonymous
          </option>
          <option
            value="BASIC"
          >
            Basic
          </option>
        </select>
      </div>
      <div
        class="form-group"
      >
        <label
          class="form-label"
        >
          username
           
          <span
            class="red-text"
          >
            *
          </span>
        </label>
        <small
          class="form-text text-muted"
        >
          description.osiPi.username
        </small>
        <input
          class="form-control"
          id="osiPi_username"
          placeholder="placeholder.osiPi.username"
          required=""
          type="text"
          value="mock-user"
        />
        <div
          class="invalid-feedback"
        />
      </div>
      <div
        class="form-group"
      >
        <label
          class="form-label"
        >
          password
           
          <span
            class="red-text"
          >
            *
          </span>
        </label>
        <small
          class="form-text text-muted"
        >
          description.osiPi.password
        </small>
        <input
          class="form-control"
          id="osiPi_password"
          placeholder="placeholder.osiPi.password"
          required=""
          type="password"
          value="mock-password"
        />
        <div
          class="invalid-feedback"
        />
      </div>
    </div>
  </div>
  <div
    class="row"
  >
    <div
      class="col"
    >
      <span>
         
      </span>
    </div>
  </div>
  <div
    class="card"
  >
    <div
      class="card-body"
    >
      <div
        class="card-subtitle h6"
      >
        query.config
      </div>
      <div
        class="row"
      >
        <div
          class="col"
        >
          <span>
             
          </span>
        </div>
      </div>
      <div
        class="form-group"
      >
        <label
          class="form-label"
        >
          request.frequency
           
          <span
            class="red-text"
          >
            *
          </span>
        </label>
        <small
          class="form-text text-muted"
        >
          description.osiPi.requestFrequency
        </small>
        <input
          class="form-control"
          id="osiPi_requestFrequency"
          placeholder="placeholder.osiPi.requestFrequency"
          required=""
          type="text"
          value="5"
        />
        <div
          class="invalid-feedback"
        />
      </div>
      <div
        class="form-group"
      >
        <label
          class="form-label"
        >
          catchup.frequency
           
          <span
            class="red-text"
          >
            *
          </span>
        </label>
        <small
          class="form-text text-muted"
        >
          description.osiPi.catchupFrequency
        </small>
        <input
          class="form-control"
          id="osiPi_catchupFrequency"
          placeholder="placeholder.osiPi.catchupFrequency"
          required=""
          type="text"
          value="0.1"
        />
        <div
          class="invalid-feedback"
        />
      </div>
      <div
        class="form-group"
      >
        <label
          class="form-label"
        >
          max.request.duration
           
          <span
            class="red-text"
          >
            *
          </span>
        </label>
        <small
          class="form-text text-muted"
        >
          description.osiPi.maxRequestDuration
        </small>
        <input
          class="form-control"
          id="osiPi_maxRequestDuration"
          placeholder="placeholder.osiPi.maxRequestDuration"
          required=""
          type="text"
          value="60"
        />
        <div
          class="invalid-feedback"
        />
      </div>
      <div
        class="form-group"
      >
        <label
          class="form-label"
        >
          query.offset
           
          <span
            class="red-text"
          >
            *
          </span>
        </label>
        <small
          class="form-text text-muted"
        >
          description.osiPi.queryOffset
        </small>
        <input
          class="form-control"
          id="osiPi_queryOffset"
          placeholder="placeholder.osiPi.queryOffset"
          required=""
          type="text"
          value="0"
        />
        <div
          class="invalid-feedback"
        />
      </div>
    </div>
  </div>
  <div
    class="row"
  >
    <div
      class="col"
    >
      <span>
         
      </span>
    </div>
  </div>
  <div
    class="card"
  >
    <div
      class="card-body"
    >
      <div
        class="card-subtitle h6"
      >
        tags
         
        tags.explanation
      </div>
      <div
        class="row"
      >
        <div
          class="col"
        >
          <span>
             
          </span>
        </div>
      </div>
      <div
        class="form-group"
      >
        <label
          class="form-label"
        >
          read.tags
        </label>
        <textarea
          class="form-control"
          id="osiPiTags"
          placeholder="placeholder.osiPi.tags"
        >
          tag1
tag2
        </textarea>
        <div
          class="invalid-feedback"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`renders the default OsiPiForm 1`] = `
<div>
  <div
    class="form-group"
  >
    <label
      class="form-label"
    >
      api.url
       
      <span
        class="red-text"
      >
        *
      </span>
    </label>
    <small
      class="form-text text-muted"
    >
      description.osiPi.apiUrl
    </small>
    <input
      class="form-control"
      id="osiPi_apiUrl"
      placeholder="placeholder.osiPi.apiUrl"
      required=""
      type="text"
      value=""
    />
    <div
      class="invalid-feedback"
    />
  </div>
  <div
    class="form-group"
  >
    <div
      class="form-check form-check-inline"
    >
      <input
        checked=""
        class="form-check-input"
        id="osiPi_verifySSL"
        type="checkbox"
      />
      <label
        class="form-check-label"
        for="osiPi_verifySSL"
        title=""
      >
        verify.ssl
      </label>
    </div>
  </div>
  <div
    class="form-group"
  >
    <label
      class="form-label"
    >
      server.name
       
      <span
        class="red-text"
      >
        *
      </span>
    </label>
    <small
      class="form-text text-muted"
    >
      description.osiPi.serverName
    </small>
    <input
      class="form-control"
      id="osiPi_serverName"
      placeholder="placeholder.osiPi.serverName"
      required=""
      type="text"
      value=""
    />
    <div
      class="invalid-feedback"
    />
  </div>
  <div
    class="card"
  >
    <div
      class="card-body"
    >
      <div
        class="card-subtitle h6"
      >
        authentication
      </div>
      <div
        class="row"
      >
        <div
          class="col"
        >
          <span>
             
          </span>
        </div>
      </div>
      <div
        class="form-group"
      >
        <label
          class="form-label"
        >
          auth.mode
           
          <span
            class="red-text"
          >
            *
          </span>
        </label>
        <small
          class="form-text text-muted"
        >
          description.osiPi.authMode
        </small>
        <select
          class="form-control"
          id="osiPi_authMode"
        >
          <option
            value="ANONYMOUS"
          >
            Anonymous
          </option>
          <option
            value="BASIC"
          >
            Basic
          </option>
        </select>
      </div>
    </div>
  </div>
  <div
    class="row"
  >
    <div
      class="col"
    >
      <span>
         
      </span>
    </div>
  </div>
  <div
    class="card"
  >
    <div
      class="card-body"
    >
      <div
        class="card-subtitle h6"
      >
        query.config
      </div>
      <div
        class="row"
      >
        <div
          class="col"
        >
          <span>
             
          </span>
        </div>
      </div>
      <div
        class="form-group"
      >
        <label
          class="form-label"
        >
          request.frequency
           
          <span
            class="red-text"
          >
            *
          </span>
        </label>
        <small
          class="form-text text-muted"
        >
          description.osiPi.requestFrequency
        </small>
        <input
          class="form-control"
          id="osiPi_requestFrequency"
          placeholder="placeholder.osiPi.requestFrequency"
          required=""
          type="text"
          value="5"
        />
        <div
          class="invalid-feedback"
        />
      </div>
      <div
        class="form-group"
      >
        <label
          class="form-label"
        >
          catchup.frequency
           
          <span
            class="red-text"
          >
            *
          </span>
        </label>
        <small
          class="form-text text-muted"
        >
          description.osiPi.catchupFrequency
        </small>
        <input
          class="form-control"
          id="osiPi_catchupFrequency"
          placeholder="placeholder.osiPi.catchupFrequency"
          required=""
          type="text"
          value="0.1"
        />
        <div
          class="invalid-feedback"
        />
      </div>
      <div
        class="form-group"
      >
        <label
          class="form-label"
        >
          max.request.duration
           
          <span
            class="red-text"
          >
            *
          </span>
        </label>
        <small
          class="form-text text-muted"
        >
          description.osiPi.maxRequestDuration
        </small>
        <input
          class="form-control"
          id="osiPi_maxRequestDuration"
          placeholder="placeholder.osiPi.maxRequestDuration"
          required=""
          type="text"
          value="60"
        />
        <div
          class="invalid-feedback"
        />
      </div>
      <div
        class="form-group"
      >
        <label
          class="form-label"
        >
          query.offset
           
          <span
            class="red-text"
          >
            *
          </span>
        </label>
        <small
          class="form-text text-muted"
        >
          description.osiPi.queryOffset
        </small>
        <input
          class="form-control"
          id="osiPi_queryOffset"
          placeholder="placeholder.osiPi.queryOffset"
          required=""
          type="text"
          value="0"
        />
        <div
          class="invalid-feedback"
        />
      </div>
    </div>
  </div>
  <div
    class="row"
  >
    <div
      class="col"
    >
      <span>
         
      </span>
    </div>
  </div>
  <div
    class="card"
  >
    <div
      class="card-body"
    >
      <div
        class="card-subtitle h6"
      >
        tags
         
        tags.explanation
      </div>
      <div
        class="row"
      >
        <div
          class="col"
        >
          <span>
             
          </span>
        </div>
      </div>
      <div
        class="form-group"
      >
        <label
          class="form-label"
        >
          read.tags
        </label>
        <textarea
          class="form-control"
          id="osiPiTags"
          placeholder="placeholder.osiPi.tags"
        />
        <div
          class="invalid-feedback"
        />
      </div>
    </div>
  </div>
</div>
`;
